src/main/java/co/inventorsoft/academy/springboottask/dao/EmailDAO.java

```diff
@@ -0,0 +1,115 @@
+package co.inventorsoft.academy.springboottask.dao;
+
+import co.inventorsoft.academy.springboottask.model.Email;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Repository;
+import org.springframework.util.FileSystemUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+@Repository
+public class EmailDAO {
+    private String emailsLocation;
```

Comment:
You don't need to access file every time. Just read it on boot time and store it inside collection. Then on shutdown write you collection content into file once. You can use bean lifecycle events for this